lib/manageiq/session_store.rb

```diff
@@ -0,0 +1,88 @@
+module ManageIQ
+  module SessionStore
+    def self.store
+      @store
+    end
+
+    def self.store=(session_store)
+      adapter = case session_store
+                when :active_record_store then ActiveRecordStoreAdapter.new
+                when :memory_store        then MemoryStoreAdapter.new
+                when :mem_cache_store     then MemCacheStoreAdapter.new
+                else
+                  raise "Please provide a valid Session Store"
+                end
+
+      adapter.patch_session_store_class
+      set_store_config(adapter)
+
+      @store = adapter.class_to_patch.new(nil, adapter.session_options)
+    end
+
+    def self.set_store_config(adapter)
+      Vmdb::Application.config.session_store(*adapter.session_store_config_args)
+      msg = "Using session_store: #{Vmdb::Application.config.session_store}"
+      $log.info("MIQ(SessionStore) #{msg}")
+      puts "** #{msg}" if !Rails.env.production? && adapter.type != :mem_cache_store
+    end
+
+    def self.revoke_all(session_ids)
```

Comment:
I wonder if the method should just be called `delete_sessions`, just to avoid creating 2 terms for the same thing.

Reply:
As mentioned above to Oleg:

https://github.com/ManageIQ/manageiq/pull/20549#issuecomment-692840586

If you want a name change, that is fine, but please make a decision and stick with it, since others are planning on working with the changes from this PR to start on their own work (Keenan: Redis store, Libor:  API endpoint).

Reply:
Fair enough...I just threw the name `revoke_all` out in the design, but it wasn't a hard decision.  If you think delete_sessions makes more sense, then we can go with that.  Otherwise, let's change the patched method to be revoke_all.  My comment here was really about keeping the names consistent.

@lpichler @gtanzillo @kbrock Any concerns on the interface name?  Any of `revoke`, `revoke_all`, `delete_sessions` works for me.

Reply:
I'm 👍 for `revoke_all` but I'll admit, I don't have a strong opinion on it.

Reply:
Left some of my opinions in one of Keenan's comments, since it is related:

https://github.com/ManageIQ/manageiq/pull/20549#discussion_r489534979

Reply:
Since there was a bunch of "¯\\\_(ツ)\_/¯" from everyone, I ended up going with:

- Renaming the interface class `ManageIQ::Session` (instead of "`SessionStore`")
- Naming the method `revoke`

So it is called like `ManageIQ::Session.revoke`.  The `@store` will call `delete_all` for the configured session store based on my previous rational here:

https://github.com/ManageIQ/manageiq/pull/20549#discussion_r489534979